python/triton/language/core.py

```diff
@@ -499,67 +499,67 @@ def __index__(self):
         return self.value
 
     def __add__(self, other):
-        return constexpr(self.value + other.value)
+        return constexpr(self.value + _constexpr_to_value(other))
```

Comment:
I'm confused why these are needed only for interpreter mode.  Could you add a comment to the code?

Reply:
Is the same reason as described above.

Reply:
I feel like a comment on just one of them that says "here and below" would be fine, it's easy to see.